### ISSUE TITLE
Agenda poor performance 

### ISSUE THREAD

**Issue by francesco-clementi-92** [wix/react-native-calendars#1985]
## Description
I'd like to reopen this issue: #1088 

It should be reopened as the issue is still present, even with the last version. 

I keep getting: 

```
VirtualizedList: You have a large list that is slow to update - make sure your renderItem function renders components that follow React performance best practices like PureComponent, shouldComponentUpdate, etc. {"contentLength": 3428, "dt": 1483, "prevDt": 1470}
```

## Environment

Please run these commands in the project folder and fill in their results:

* `react-native-calendars@1.1287.0`:
* `@react-native-async-storage/async-storage@1.17.7
│ └── react-native@0.69.3 deduped
├─┬ @react-native-firebase/app@15.1.1
│ └── react-native@0.69.3 deduped
├─┬ @react-native-google-signin/google-signin@8.0.0
│ └── react-native@0.69.3 deduped
├─┬ @react-navigation/bottom-tabs@6.3.1
│ ├─┬ @react-navigation/elements@1.3.3
│ │ └── react-native@0.69.3 deduped
│ └── react-native@0.69.3 deduped
├─┬ @react-navigation/native-stack@6.6.2
│ └── react-native@0.69.3 deduped
├─┬ @react-navigation/native@6.0.10
│ └── react-native@0.69.3 deduped
├─┬ react-native-bootsplash@4.2.3
│ └── react-native@0.69.3 deduped
├─┬ react-native-calendars@1.1287.0
│ └─┬ recyclerlistview@3.0.5
│   └── react-native@0.69.3 deduped
├─┬ react-native-keyboard-aware-scroll-view@0.9.5
│ ├─┬ react-native-iphone-x-helper@1.3.1
│ │ └── react-native@0.69.3 deduped
│ └── react-native@0.69.3 deduped
├─┬ react-native-modal@13.0.1
│ └── react-native@0.69.3 deduped
├─┬ react-native-paper@4.12.2
│ └── react-native@0.69.3 deduped
├─┬ react-native-push-notification@8.1.1
│ └── react-native@0.69.3 deduped
├─┬ react-native-safe-area-context@4.3.1
│ └── react-native@0.69.3 deduped
├─┬ react-native-screens@3.14.0
│ └── react-native@0.69.3 deduped
└─┬ react-native@0.69.3
  └─┬ @react-native-community/cli@8.0.5
    └── react-native@0.69.3 deduped`:

**Comment by sfm999** [comment]
Yep, same here. I think it's to do with the months updating during the scroll.
I also noticed that the futureScrollRange and pastScrollRange don't work, at least not on my samsung s21+.

**Comment by khiet-vo** [comment]
Same problem 2024 :(
